catalystwan/api/device_action_api.py

```diff
@@ -115,6 +116,87 @@ def wait_for_come_up():
         wait_for_come_up()
 
 
+class PatchUpgradeAction(DeviceActionAPI):
+    """API method to perform reboot on Device."""
+
+    def execute(self, patch_version: str = "default"):
+        """Patches vmanage
+
+        Returns:
+            patching process id
+
+        Raises:
+            Exception when patching times out.
+
+        Usage example:
+        # Create session and chose vmanage device
+        session = create_manager_session(...)
+        device = DevicesAPI(session).get().filter(personality = Personality.VMANAGE)[0]
+        # Patch device. Requires passing the vmanage version vmanage will be patched to.
+        PatchAction(session, device).execute(version)
+        """
+        if self.dev.personality != Personality.VMANAGE:
+            raise Exception(f"Patch upgrade cannot be executed for {self.dev.personality}")
+
+        body = {
+            "action": "patchupgrade",
+            "deviceType": "vmanage",
+            "patchversion": patch_version,
+            "devices": [{"deviceIP": self.dev.id, "deviceId": self.dev.uuid}],
+        }
+
+        response = self.session.post("/dataservice/device/action/patchupgrade", json=body).json()
+        if response.get("id"):
+            self.action_id = response["id"]
+        else:
+            raise Exception(f"Problem with patching {self.dev.id} occurred")
+
+    def wait_for_completed(
```

Comment:
waiting for task already implemented so it can be reused for any task in `catalystwan.api.task_status_api.Task`
```python
from catalystwan.api.task_status_api import Task
task = Task(session, task_id)
task.wait_for_completed()
```
Unfortunately current assumption in SDK is that session is not being interrupted (eg. by reboot)
I suggest detect reboot and wait until vManage is ready outside SDK - then use `Task` from SDK.